tests/knowledge_base/test_actions.py

```diff
@@ -116,12 +156,20 @@ def compare_slots(slot_list_1, slot_list_2):
         ),
     ],
 )
-async def test_action_run(data_file, slots, expected_slots):
+async def test_action_run(data_file, latest_message, slots, expected_slots):
     knowledge_base = InMemoryKnowledgeBase(data_file)
     action = ActionQueryKnowledgeBase(knowledge_base)
 
     dispatcher = CollectingDispatcher()
-    tracker = Tracker("default", slots, {}, [], False, None, {}, "action_listen")
+
+    tracker = Tracker(
+        "default", slots, latest_message, [], False, None, {}, "action_listen"
+    )
+
+    # To prevent unintended modifications, create a copy of the slots dictionary
+    # before passing it to the Tracker object, as dictionaries in
+    # Python are mutable and passed by reference.
+    initial_slots = dict(slots)
```

Comment:
Shouldn't you use `copy.deepcopy` instead of casting? I think the casting won't prevent mutability issues 🤔

Reply:
Thanks for the feedback Anca Lita. Incorporated your suggestion